x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/inference_index_template.json

```diff
@@ -2,7 +2,7 @@
   "order" : 0,
   "version" : ${xpack.ml.version.id},
   "index_patterns" : [
-    ".ml-inference-000002"
+    ".ml-inference-000003"
```

Comment:
Are these templates automatically installed when any new node comes online? Or does the new node have to be a master node?

@dimitris-athanasiou do you know?

Reply:
Aside of the question, I believe the index pattern here should be `.ml-inference-*` and we shouldn't have to change this each time.

Back to the question, all nodes get the templates installed regardless of whether they're master or not.

Reply:
@dimitris-athanasiou I am not sure. What if an older node has a specific mapping in mind and tries to write to the index?

A new index would be created with the new mapping (if it didn't already exist) and the new node does not have code to support the new mapping. 

Keeping it versioned allows only NEW nodes to reference the new index mapping.

Reply:
I see, it makes sense.